### DISCUSSION TITLE
Numbered Headings in Callout Boxes Trigger Unintended Numbered Lists in Typst Format

### DISCUSSION THREAD

**Discussion by produnis** [quarto-dev/quarto-cli#11575] [Q&A]
### Description

When creating a heading inside a callout box in Typst format with `Quarto 1.6.37`, if the heading starts with a number (e.g., `## 1. Heading`), Typst interprets this as the start of a numbered list. This results in a line break after the heading, and a numbered list begins on the next line.
```

---
title: test
format: typst
keep-md: true
keep-typ: true
---

::: {.callout-note appearance="default"}
## FC Schalke 04 (famous german football club)
We are the best
:::



::: {.callout-note appearance="minimal"}
## 1. FC Kaiserslautern (another german football club)
We are not
:::

```

renders to

![](https://i.imgur.com/kh743cM.jpeg)

I am wondering if this is intended behavior.

Also, I notices that it takes no effect if you specify the callout with the `appearance` parameter, as `appearance="default"` ,`appearance="simple"` and  `appearance="minimal"` render the full "default" callout-box.

**Comment by cscheid** [comment]
This is definitely not intended behavior.

**Reply by cscheid** [reply]
This is going to be... impossible? for Quarto to fix automatically. The `keep-typ: true` output offers a hint of what's going on. The callout is invoked as a typst function call:

```typst
#callout(
body: 
[
We are not

]
, 
title: 
[
1. FC Kaiserslautern (another german football club)
]
, 
background_color: 
rgb("#dae6fb")
, 
icon_color: 
rgb("#0758E5")
, 
icon: 
fa-info()
)
```

In this context, the title is just "content", and nothing else. Typst (understandably, if counter-intuitively) converts that to an enumeration. The unfortunate part is that Typst knows not to do it when the title _text_ is inside a heading. Consider this pure-typst example:

```typst
== 1. This is safe.

Some stuff

#heading(level: 2, [1. This is not.])
```

The output you get is less offensively-wrong than the callout output we generate, but the "bug" is there if you pay close attention to the spacing:

<img width="137" alt="image" src="https://github.com/user-attachments/assets/34b8a81c-a167-47b9-b369-4f6fea695696">

I think the solution here is to escape the `.`:

```typst
== 1\. This is safe.

Some stuff

#heading(level: 2, [1\. This is not.])
```

<img width="134" alt="image" src="https://github.com/user-attachments/assets/5f665ca1-5f45-49c4-a99d-c2dc936b96e2">

It's indeed quite unfortunate that the issue only manifests in Typst output.

**Reply by produnis** [reply]
Thx or the workaround!

But how could I make  callout boxes in typst care for the appearance parameter?

**Reply by cderv** [reply]
> But how could I make callout boxes in typst care for the appearance parameter?

I am not sure typst support `appearance` configuration for callout yet 🤔

**Reply by mcanouil** [reply]
FYI

- https://github.com/quarto-dev/quarto-cli/issues/9832
